src/System.Diagnostics.Process/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -67,16 +69,12 @@ internal static PerformanceCounterLib GetPerformanceCounterLib(string machineNam
             else
                 machineName = machineName.ToLowerInvariant();
 
-            LazyInitializer.EnsureInitialized(ref s_libraryTable, ref s_internalSyncObject, () => new Dictionary<string, PerformanceCounterLib>());
+            LazyInitializer.EnsureInitialized(ref s_libraryTable, ref s_internalSyncObject, () => new ConcurrentDictionary<string, PerformanceCounterLib>());
 
             string libraryKey = machineName + ":" + lcidString;
-            PerformanceCounterLib library;
-            if (!PerformanceCounterLib.s_libraryTable.TryGetValue(libraryKey, out library))
-            {
-                library = new PerformanceCounterLib(machineName, lcidString);
-                PerformanceCounterLib.s_libraryTable[libraryKey] = library;
-            }
-            return library;
+
+	        return PerformanceCounterLib.s_libraryTable.GetOrAdd(libraryKey,
+		        (k) => new PerformanceCounterLib(machineName, lcidString));
```

Comment:
This is going to allocate a closure and a delegate every time this method is called, due to capturing machineName and lcidString.  This should instead use the overload that takes a factory argument:
https://github.com/dotnet/corefx/blob/master/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L1010
to which you can pass a value tuple `(machineName, lcidString)` to avoid both of those.